lib/main.js

```diff
@@ -172,83 +164,93 @@ module.exports = class Router extends EventEmitter {
         };
     }
 
-    postBundleParseCallback() {
-        return (req, res, next) => {
-            body(req, res, {}, (error, bodyObj) => {
-                if (error) {
-                    return next(
-                        boom.boomify(error, {
-                            statusCode: 400,
-                            message: 'Failed parsing postBundle POST-body',
-                        })
-                    );
+    postBundleHandler(type) {
+        assert(
+            ALLOWED_TYPES.includes(type),
+            `Expected type to 1 of ${ALLOWED_TYPES.join(', ')}, but got ${type}`
+        );
+        return async (req, res, next) => {
+            let feedIds;
+            try {
+                const payload = await body(req, res, {});
+                const result = schemas.ids.validate(payload);
+                if (result.error) {
+                    throw result.error;
                 }
-                res.locals.payload = bodyObj;
-                next();
-            });
-        };
-    }
+                feedIds = result.value;
+            } catch (e) {
+                return next(
+                    boom.boomify(e, {
+                        statusCode: 400,
+                        message: 'Invalid POST-body',
+                    })
+                );
+            }
 
-    postBundleValidateCallback() {
-        return (req, res, next) => {
-            schemas.ids.validate(res.locals.payload, (error, payload) => {
-                if (error) {
+            try {
+                const response = await this.bundle({
+                    feedIds,
+                    uri: this.buildUri('bundle', req.headers.host, req.secure),
+                    type,
+                });
+                this.emit(
+                    'request success',
+                    res.locals.track,
+                    req.method,
+                    req.path,
+                    response.file
+                );
+                res.json(response);
+            } catch (e) {
+                this.logger.error('Failed creating bundle', e);
+                if (e.message.includes('No file with name')) {
```

Comment:
I think we should probably have a better way to handle errors bubbling up from the sinks. Worth ignoring for now to get this out and revisiting afterwards though.